configure.ac

```diff
@@ -670,8 +673,20 @@ AC_PROG_LD_GNU
 
 AC_CHECK_HEADERS(sys/filio.h sys/sockio.h netdb.h utime.h sys/utime.h semaphore.h sys/un.h linux/rtc.h sys/syscall.h sys/mkdev.h sys/uio.h sys/param.h sys/sysctl.h libproc.h sys/prctl.h copyfile.h)
 AC_CHECK_HEADERS(sys/param.h sys/socket.h sys/ipc.h sys/utsname.h alloca.h ucontext.h pwd.h sys/select.h netinet/tcp.h netinet/in.h unistd.h sys/types.h link.h asm/sigcontext.h sys/inotify.h arpa/inet.h complex.h unwind.h)
-AC_CHECK_HEADER(unistd.h, [HAVE_UNISTD_H=1], [HAVE_UNISTD_H=0])
+
+# zlib/configure checks for stdarg.h existance and defines HAVE_STDARG_H on compiler
+# command line (not config.h) if it is present.
+CFLAGS="$CFLAGS -DHAVE_STDARG_H"
```

Comment:
I think we should put all the zlib defines into `ZLIB_EXTRA_CFLAGS` (or whatever name you like) and pass them to the compiler in `support/Makefile.am` using a target specific cflags argument.   I don't think we should put a bunch of stuff into the global CFLAGS just because zlib don't use a config file.

Reply:
That creates a dependency on including config.h before zlib.h.
Which appears to already be met but previously wasn't required.
Or knowing which files use zlib and using ZLIB_EXTRA_CFLAGS only on them.

I was aiming for a simple consistency, at the cost of possibly unused options.
Though on the other hand, I didn't want to add `#include config.h` to zlib either, giving it even more defines, in case they risky.

We do have the same `HAVE_STDARG_H` as them, so it kinda isn't pollution -- compiler command line vs. config.h, but shared valid semantic.

The stdarg.h one I can just remove it appears.

Mono misuses CFLAGS also as I understand.
It is supposed to be for the user, like -O0 -g, not things mono actually requires.

Reply:
Oh and the `HAVE_HIDDEN` and `_LARGEFILE64_SOURCE`. `_LARGEFILE64_SOURCE` seems like possible trouble either way, I'd have to read up on it again.